app/uk/gov/hmrc/exports/models/declaration/AdditionalDeclarationType.scala

```diff
@@ -30,6 +32,12 @@ object AdditionalDeclarationType extends Enumeration {
       Writes(v => JsString(v.toString))
     )
 
+  def fromString(str: String): Try[AdditionalDeclarationType] = Try {
+    AdditionalDeclarationType.values
+      .find(_.toString == str)
+      .getOrElse(throw new IllegalArgumentException(s"$str is not a valid value for AdditionalDeclarationType"))
```

Comment:
Could not this be returning an `Either` instead of a `Try` in order to avoid to raise an exception?

Reply:
This is not part of the parsers, so it should not rely on `XmlParserResult`. And returning `Either` here seems to me a bit like over-engineering.